network/p2p/libp2pNode.go

```diff
@@ -528,32 +621,49 @@ func bootstrapLibP2PHost(ctx context.Context,
 	// gather all the options for the libp2p node
 	options := []config.Option{
 		libp2p.ListenAddrs(sourceMultiAddr), // set the listen address
-		libp2p.Identity(key),                // pass in the networking key
-		libp2p.ConnectionManager(conMgr),    // set the connection manager
+		libp2p.Identity(libp2pKey),          // pass in the networking key
 		transport,                           // set the protocol
-		libp2p.Ping(true),                   // enable ping
 	}
 
-	// if allowlisting is enabled, create a connection gator with allowListAddrs
-	if allowList {
-		// create a connection gater
-		connGater = newConnGater(logger)
-
-		// provide the connection gater as an option to libp2p
-		options = append(options, libp2p.ConnectionGater(connGater))
-	}
+	return options, nil
+}
 
-	// create the libp2p host
-	libP2PHost, err := libp2p.New(ctx, options...)
+// DefaultPubSub returns initializes and returns a GossipSub object for the given libp2p host and options
+func DefaultPubSub(ctx context.Context, host host.Host, psOption ...pubsub.Option) (*pubsub.PubSub, error) {
+	// Creating a new PubSub instance of the type GossipSub with psOption
+	pubSub, err := pubsub.NewGossipSub(ctx, host, psOption...)
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("could not create libp2p host: %w", err)
+		return nil, fmt.Errorf("could not create libp2p gossipsub: %w", err)
 	}
+	return pubSub, nil
+}
 
-	// Creating a new PubSub instance of the type GossipSub with psOption
-	ps, err := pubsub.NewGossipSub(ctx, libP2PHost, psOption...)
-	if err != nil {
-		return nil, nil, nil, fmt.Errorf("could not create libp2p pubsub: %w", err)
+// PubsubOption generates a libp2p pubsub.Option from the given context and host
+type PubsubOption func(ctx context.Context, host host.Host) (pubsub.Option, error)
+
+func DefaultPubsubOptions(maxPubSubMsgSize int) []PubsubOption {
+	pubSubOptionFunc := func(option pubsub.Option) PubsubOption {
+		return func(_ context.Context, _ host.Host) (pubsub.Option, error) {
+			return option, nil
+		}
+	}
+	return []PubsubOption{
+		// skip message signing
+		pubSubOptionFunc(pubsub.WithMessageSigning(false)),
+		// skip message signature
+		pubSubOptionFunc(pubsub.WithStrictSignatureVerification(false)),
+		// set max message size limit for 1-k PubSub messaging
+		pubSubOptionFunc(pubsub.WithMaxMessageSize(maxPubSubMsgSize)),
+		// no discovery
 	}
+}
 
-	return libP2PHost, connGater, ps, nil
+func WithDHTDiscovery(option ...dht.Option) PubsubOption {
+	return func(ctx context.Context, host host.Host) (pubsub.Option, error) {
+		dhtDiscovery, err := NewDHT(ctx, host, option...)
```

Comment:
For clients (unstaked nodes), i wonder if we want to use `BootstrapPeers` option to explicitly list the upstream staked nodes: https://pkg.go.dev/github.com/libp2p/go-libp2p-kad-dht#BootstrapPeers

Reply:
❤️ hey i didn't knew that there was such an option..but they should attempt a connection immediately before subscribing to any topic, else like I mentioned earlier - not all peers are discovered.

Reply:
Right, but we should probably also include bootstrap peers too, in case our routing table becomes epty:

> BootstrapPeers configures the bootstrapping nodes that we will connect to to seed and refresh our Routing Table if it becomes empty.

Reply:
made that change, passing in the bootstrap nodes to the dht for the unstaked AN